src/festim/helpers.py

```diff
@@ -17,11 +21,257 @@ def as_fenics_constant(
+        function_space: The function space to interpolate function over
+        mesh: the mesh of the domain
+        t: the time
+        temperature: the temperature
+
+    Returns:
+        fenics interpolation expression, fenics function
+    """
+
+    mapped_function = as_mapped_function(
+        value=value, mesh=mesh, t=t, temperature=temperature
+    )
+
+    fenics_interpolation_expression = fem.Expression(
+        mapped_function,
+        function_space.element.interpolation_points(),
+    )
+
+    fenics_object = fem.Function(function_space)
+    fenics_object.interpolate(fenics_interpolation_expression)
+
+    return fenics_interpolation_expression, fenics_object
+
+
+class Value:
+    """
+    A class to handle input values from users and convert them to a relevent fenics
+    object
+
+    Args:
+        input_value: The value of the user input
+
+    Attributes:
+        input_value : The value of the user input
+        fenics_interpolation_expression : The expression of the user input that is used
+            to update the `fenics_object`
+        fenics_object : The value of the user input in fenics format
+
+    """
+
+    input_value: (
+        int
+        | float
+        | np.ndarray
+        | Callable[[np.ndarray], np.ndarray]
+        | Callable[[np.ndarray, float], np.ndarray]
+        | Callable[[float], float]
+        | fem.Constant
+        | fem.Expression
+        | ufl.core.expr.Expr
+        | fem.Function
+    )
+
+    ufl_expression: ufl.core.expr.Expr
+    fenics_interpolation_expression: fem.Expression
+    fenics_object: fem.Function | fem.Constant | ufl.core.expr.Expr
+
+    def __init__(self, input_value):
+        (self,)
+        self.input_value = input_value
+
+        self.ufl_expression = None
+        self.fenics_interpolation_expression = None
+        self.fenics_object = None
+
+    def __repr__(self) -> str:
+        return str(self.input_value)
+
+    @property
+    def input_value(self):
+        return self._input_value
+
+    @input_value.setter
+    def input_value(self, value):
+        if value is None:
+            self._input_value = value
+        elif isinstance(
+            value,
+            float
+            | int
+            | fem.Constant
+            | np.ndarray
+            | fem.Expression
+            | ufl.core.expr.Expr
+            | fem.Function,
+        ):
+            self._input_value = value
+        elif callable(value):
+            self._input_value = value
+        else:
+            raise TypeError(
+                "Value must be a float, int, fem.Constant, np.ndarray, fem.Expression,"
+                f" ufl.core.expr.Expr, fem.Function, or callable not {value}"
+            )
+
+    @property
+    def time_dependent(self) -> bool:
+        """Returns true if the value given is time dependent"""
+        if self.input_value is None:
+            return False
+        if isinstance(self.input_value, fem.Constant | ufl.core.expr.Expr):
```

Comment:
What do you imply by time-dependent here? Is it only if it is a `Callable`-function with `t` as an input argument? as expression and constants can be time-dependent, as they depend on variables that could be updated over time.

Reply:
Yeah its just there to detect if t is given in the arguments of the callable, do you know of a way to detect if a `fem.Constant` or `fem. Expression` is time dependent?

Reply:
I don't think there's a way to do this, FEniCS has no concept of time.

Reply:
We could add some overloaded version of constant that we annotate this in

Reply:
We decided that this should be called `explicit_time_dependency`.